### ISSUE TITLE
Change ManageEngine CVE-2022-47966 to be more stealthy

### ISSUE THREAD

**Issue by bwatters-r7** [rapid7/metasploit-framework#17641]
The three ManageEngine modules using CVE-2022-47966 all invoke GetRuntime().Exec().  Jacob put up a blog post with alternative suggestions here: https://vulncheck.com/blog/cve-2022-47966-payload

**Comment by errorxyz** [comment]
Hey! So I've studied a few other exploits using java payloads and I think we can use metasploit's java payload in the jar form since the jar files should bypass those filters. We could initially send a payload containing java code that loads our metasploit payload from srvhost(using `URLClassLoader`) and then run our payload from there. I'm confused about one thing though- should I run the metasploit payload in a new thread or in the same thread?

Regarding the `post` invocation - I'm not really sure what you meant by "mixing that up", it would be really useful if you could elaborate on that.

Once we finish this up we could finally tackle Velociraptor's approach that just seems like a simple delete function.

Any corrections/suggestions would be appreciated since I'm still getting familiar with the codebase.

**Comment by errorxyz** [comment]
@bwatters-r7 can you assign this issue to me?

**Comment by errorxyz** [comment]
Hi! I've successfully been able to get a shell using the `java/shell_reverse_tcp` payload(none of the other payloads work). But I've noticed that the payload again uses `getRuntime().exec()` to run commands and invokes a new process `sh` under the parent `java` process to spawn a shell- so we don't really bypass the sigma rule(although we've bypassed the suricata rule). Is there any way around this problem?

**Comment by errorxyz** [comment]
@bwatters-r7 could you please look into the PR and provide some feedback if possible? Thanks.

**Comment by errorxyz** [comment]
I don't think this issue has been completely resolved, there are two more modules that need to be modified to use the payload.

**Comment by bwatters-r7** [comment]
OK; I reopened it; it auto closed based on the string `fixes #17641` in the description of https://github.com/rapid7/metasploit-framework/pull/18515

**Comment by errorxyz** [comment]
Im unable to reproduce the environment for the [manageengine_adselfservice_plus](https://github.com/rapid7/metasploit-framework/blob/master/modules/exploits/multi/http/manageengine_adselfservice_plus_saml_rce_cve_2022_47966.rb) module due to hardware limitations, however I've modified the exploit to add the java target so if anyone can verify the exploit, it'd be very useful (https://github.com/errorxyz/metasploit-framework/tree/manageengine)
setup: https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/exploit/multi/http/manageengine_adselfservice_plus_saml_rce_cve_2022_47966.md

**Comment by errorxyz** [comment]
Unassigning myself for now since I'm unable to setup the environment to exploit adselfservice plus due to hardware limitations, however I've modified the exploit for adselfservice plus, so if anyone's willing to work on it, happy to share it.
